docs: include new SvelteKit options (#32)
vite-pwa/docs: frameworks/sveltekit.md

:::

::: tip
If you're using `0.1.*` version of `SvelteKitPWA`, you should remove all references to [SvelteKit service worker module](https://kit.svelte.dev/docs/service-workers) to disable it on your application.
:::

## Generate Custom Service Worker

> From version `0.2.0`, `SvelteKitPWA` plugin will delegate your custom service worker build to `SvelteKit`, and so, you can use `service-worker` as the name of your service worker file (VanillaJS or TypeScript).

Reviewer: Is this required or optional? If required then we should change as below:
```suggestion
From version `0.2.0`, `SvelteKitPWA` plugin will delegate your custom service worker build to `SvelteKit`, and so, you must use `service-worker` as the name of your service worker file (VanillaJS or TypeScript).
```

Author: uhmm: we can create `src/service-worker.js` (kit will detect it) or `src/service-worker.ts` (we need to configure in bot, kit and pwa options)

On previous versions, we cannot use `src/service-worker.js` to avoid kit to generate the corresponding bundle.

Author: If using `src/service-worker.js`, pwa plugin will configure it by default (if missing ofc):
- https://github.com/vite-pwa/sveltekit/pull/32/files#diff-c3095d5010e65c52737a98a5d618ea24049ebe90c8470752426081d70ed6e012R32
- https://github.com/vite-pwa/sveltekit/pull/32/files#diff-c3095d5010e65c52737a98a5d618ea24049ebe90c8470752426081d70ed6e012R35

You will need to exclude the service worker registration from the `SvelteKit` configuration if you're using any virtual module:
```js
// svelte.config.js
/** @type {import('@sveltejs/kit').Config} */
const config = {
  kit: {
    serviceWorker: {
      register: false
    }
  }
};
```

You can also configure a custom service worker name, `SvelteKitPWA` plugin will rename your service worker (remember to also update the service worker name in `SvelteKitPWA` plugin configuration):
```js
// svelte.config.js
/** @type {import('@sveltejs/kit').Config} */
const config = {
  kit: {
    files: {
      serviceWorker: 'src/prompt-sw.ts',
    }
  }
};
```

> You can check SvelteKit docs for more information about [Service Worker](https://kit.svelte.dev/docs/service-workers).

Reviewer: ```suggestion
You can check SvelteKit docs for more information about [service workers](https://kit.svelte.dev/docs/service-workers).
```

::: warning

> If your custom service working is importing any `workbox-*` module (`workbox-routing`, `workbox-strategies`, etc), you will need to hack Vite build process in order to remove non `ESM` special replacements from the build process (if you don't include `process.env.NODE_ENV`, the service worker will not be registered). You only need to add this entry in your Vite config file:

Reviewer: this is perhaps not required with Vite 4.2? https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md#420-beta0-2023-02-27

Author: If using Vite 4.1 we need to "hack" Vite build. 

Maybe we can keep it, and add a suggestion when Vite 4.2 released. 

I will check it using the latest beta once new minor version (0.2.0) released and published.

Author: Testing with Vite 4.2.0-beta.1 not working using injectManifest strategy, `prompt-sw.js` containing `process.env.NODE_ENV` entries.

:::

::: tip
If you're using `0.1.*` version of `SvelteKitPWA`, you should remove all references to [SvelteKit service worker module](https://kit.svelte.dev/docs/service-workers) to disable it on your application.
:::

## Generate Custom Service Worker

From version `0.2.0`, `SvelteKitPWA` plugin will delegate your custom service worker build to `SvelteKit`, and so, you can use `src/service-worker.js`: in that case, you don't need to configure the `filename` in pwa plugin options.

> You will need to exclude the service worker registration from the `SvelteKit` configuration if you're using any virtual module:

Reviewer: is it virtual modules from SvelteKit or virtual modules from vite PWA plugin or either?

> From version `0.2.0`, `SvelteKitPWA` plugin will delegate your custom service worker build to `SvelteKit`, and so, you can use `src/service-worker.js`: in that case, you don't need to configure the `filename` in pwa plugin options.
> 
> You will need to exclude the service worker registration from the `SvelteKit` configuration if you're using any virtual module:
> ```js
> // svelte.config.js
> /** @type {import('@sveltejs/kit').Config} */
> const config = {
>   kit: {
>     serviceWorker: {
>       register: false
>     }
>   }
> };
> ```
> 
> You can also configure a custom service worker, remember to also update the service worker name in `SvelteKitPWA` plugin configuration, for example:
> ```js
> // svelte.config.js
> /** @type {import('@sveltejs/kit').Config} */
> const config = {
>   kit: {
>     files: {
>       serviceWorker: 'src/prompt-sw.js', // or `src/prompt-sw.ts`
>     }
>   }
> };
> ```

Reviewer: it was a bit confusing that the name was discussed in the first and third paragraphs. easier to follow if we combine them:
```suggestion
From version `0.2.0`, `SvelteKitPWA` plugin will delegate your custom service worker build to `SvelteKit`, and so by default you will be expected to put your service worker in `src/service-worker.js`. If you would like, you can use a custom file location by changing the corresponding SvelteKit option:
```js
// svelte.config.js
/** @type {import('@sveltejs/kit').Config} */
const config = {
  kit: {
    files: {
      serviceWorker: 'src/my-sw.js', // or `src/my-sw.ts`
    }
  }
};
```

You will need to exclude the service worker registration from the `SvelteKit` configuration if you're using any virtual module:
```js
// svelte.config.js
/** @type {import('@sveltejs/kit').Config} */
const config = {
  kit: {
    serviceWorker: {
      register: false
    }
  }
};
```
```

:::

::: tip
If you're using `0.1.*` version of `SvelteKitPWA`, you should remove all references to [SvelteKit service worker module](https://kit.svelte.dev/docs/service-workers) to disable it on your application.
:::

## Generate Custom Service Worker

From version `0.2.0`, `SvelteKitPWA` plugin will delegate your custom service worker build to `SvelteKit`, and so by default you will be expected to put your service worker in `src/service-worker.js`. If you would like, you can use a custom file location by changing the corresponding SvelteKit option:
```js
// svelte.config.js
/** @type {import('@sveltejs/kit').Config} */
const config = {
  kit: {
    files: {
      serviceWorker: 'src/my-sw.js', // or `src/my-sw.ts`
    }
  }
};

export default config;
```

Then in your Vite config file:
```js
// vite.config.js or vite.config.ts
/** @type {import('vite').UserConfig} */
const config = {
  plugins: [
    sveltekit(),
    SvelteKitPWA({
      strategies: 'injectManifest',
      srcDir: 'src',

>       filename: 'my-sw.js', // or `my-sw.ts`

Author: how can I read/access kit configuration from vite?

>       filename: 'my-sw.js', // or `my-sw.ts`

Reviewer: you could use `loadSvelteConfig` from `@sveltejs/vite-plugin-svelte`

>       filename: 'my-sw.js', // or `my-sw.ts`

Author: this will be included as a new feat

   * @see https://kit.svelte.dev/docs/configuration#trailingslash
   */
  trailingSlash?: 'never' | 'always' | 'ignore'

>   /**
>    * If you're using older version of SvelteKit, you can set this to `false`.
>    *
>    * @default true
>    */
>   latestKit?: boolean

Reviewer: I think this isn't needed anymore
```suggestion
```
